ForMaL/Workshops/workshop1.lean

```diff
@@ -0,0 +1,111 @@
+/- # Basics: rewriting, calculating, reasoning forwards and backwards
+References: [MIL] Mathematics in Lean, [Tut] Tutorials project.
+Most of the demonstration section comes [MIL].
+-/
+
+import Mathlib.Tactic
+
+/-
+# Rewriting and calculating
+-/
+
+-- Lean knows some lemmas about real numbers.
```

Comment:
maybe instead of real numbers, we can just say commutative rings?

Reply:
obviously, if we do that, we would need to change all `ℝ` to some `{R : Type_}` and add the instance `[CommRing R]`

Reply:
It might be better to keep things simple in the first session, and not use any typeclasses or implicit arguments.